packages/cw721/src/state.rs

```diff
@@ -0,0 +1,241 @@
+use std::marker::PhantomData;
+
+use cosmwasm_schema::cw_serde;
+use cosmwasm_std::{Addr, BlockInfo, CustomMsg, Decimal, StdResult, Storage, Timestamp};
+use cw_ownable::{OwnershipStore, OWNERSHIP_KEY};
+use cw_storage_plus::{Index, IndexList, IndexedMap, Item, Map, MultiIndex};
+use cw_utils::Expiration;
+use serde::de::DeserializeOwned;
+use serde::Serialize;
+
+/// Creator owns this contract and can update collection info!
+/// !!! Important note here: !!!
+/// - creator is stored using using cw-ownable's OWNERSHIP singleton, so it is not stored here
+/// - in release v0.18.0 it was used for minter (which is confusing), but now it is used for creator
+pub const CREATOR: OwnershipStore = OwnershipStore::new(OWNERSHIP_KEY);
+/// - minter is stored in the contract storage using cw_ownable::OwnershipStore (same as for OWNERSHIP but with different key)
+pub const MINTER: OwnershipStore = OwnershipStore::new("collection_minter");
+
```

Comment:
`CREATOR` and `MINTER`. Please note that singleton ownership was used before for minter, but now used for creator! new minter ownership has a new key `collection_minter`. This is also covered in migration (as shown later below).

Reply:
Okay so to be clear, we now have an optional second owner which is creator. The max number of owners in the contract would now be 2 and not more than that right?

Reply:
> Okay so to be clear, we now have an optional second owner which is creator. The max number of owners in the contract would now be 2 and not more than that right?

Correct. They are both optional: creator and minter. Naming convention before was very confusing - started with the introduction of `cw-ownable`. There it was minter = owner = ownership.

Imo, using terms `creator` and `minter` is much more clear. Ownership appears here only in terms of minter ownership and creator ownership. Both uses for 2-step for transfer as defined by `cw-ownable`:
1. Owner transfers to new owner
2. New owner accepts, on acceptance minter/creator ownership is changed